### PR TITLE
cshore-merge

### PR DESCRIPTION
**Pull request by dyoung8**
the cshore-merge branch of testbedutils is the most up-to-date version of testbedutils on bitbucket that was used in the cshore-merge version of cmtb (with all the new CSHORE changes).  I dont recall any major changes that I made to this repo, other than 2-3 (i think), so hopefully this will be straightforward.

### CODE REVIEW

File: gridTools.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import numpy as np
 from matplotlib import pyplot as plt
 from scipy.interpolate import griddata
+import makenc
```
Comment (SBFRF) [inline gridTools.py]: This throws a dependency error when importing in a project that is not in the cmtb folder.  I moved to the function to remove the entire modules dependency but ideally we remove this dependency entirely by taking the netCDF writing out of the makeTimeMeanBackgroundBathy function and have it return the input to a write function wherever makeTimeMeanBackgroundBathy is used.  

Thoughts?

File: kalman_filter.py [hunk 2]
```diff
@@ -133,6 +134,163 @@ def replacecBathyMasksWithNans(dictionary):
             dictionary[var] = np.ma.filled(dictionary[var], fill_value=np.nan)
     return dictionary
 
+def cBathy_VarianceLogic(cBathy, variancePacket, rawspec, varianceThreshold=0.14, percentThreshold=0.35):
```
Comment (SBFRF) [inline kalman_filter.py]: interesting i thought this was already in the development branch.

File: sblib.py [hunk 3]
```diff
@@ -169,7 +170,7 @@ def statsBryant(observations, models):
     """
     obsNaNs = np.argwhere(np.isnan(observations)).squeeze()
     modNaNs = np.argwhere(np.isnan(models)).squeeze()
-    if isinstance(observations, np.ma.masked_array) or isinstance(models, np.ma.masked_array):
```
Comment (SBFRF) [inline sblib.py]: lets keep the is instance function, i think that's the proper way to do those comparisons, but totally open to input

File: waveLib.py [hunk 4]
```diff
@@ -441,8 +441,8 @@ def stats1D(fspec, frqbins, lowFreq=0.05, highFreq=0.5):
     df = np.diff(np.append(frqbins[0], frqbins), n=1)
 
     # truncating spectra to sea/swell band
-    # [idx, _] = sb.findbtw(frqbins, lowFreq, highFreq, type=3)
-    idx = np.argwhere((frqbins >= lowFreq) & (frqbins <= highFreq)).squeeze()
```
Comment (SBFRF) [inline waveLib.py]: please keep 

idx = np.argwhere((frqbins >= lowFreq) & (frqbins <= highFreq)).squeeze()

File: gridTools.py (hunk 1 above)
Comment (dyoung8) [reply gridTools.py]: makenc as a submodule?  especially if we start using it across projects - like a general nc creation library?

File: sblib.py (hunk 3 above)
Comment (dyoung8) [reply sblib.py]: Fixed.  this line should be identical to the deleted one above.

File: waveLib.py (hunk 4 above)
Comment (dyoung8) [reply waveLib.py]: Fixed.  these two lines should now be identical the deleted ones above.

File: gridTools.py (hunk 1 above)
Comment (dyoung8) [reply gridTools.py]: fixed.

**Comment by dyoung8** [comment]
This pull request is ready for re-review.  all review comments addresed.

**Comment by SBFRF** [comment]
Looks good, implementing new mostly minor changes to development branch

**Pull request merged**